llm_perf/common/benchmark_runner.py

```diff
@@ -0,0 +1,115 @@
+import os
+import traceback
+from abc import ABC, abstractmethod
+from itertools import product
+from logging import getLogger
+from typing import Any, Dict, List, Optional
+
+from llm_perf.common.utils import (
+    CANONICAL_PRETRAINED_OPEN_LLM_LIST,
+    OPEN_LLM_LIST,
+    PRETRAINED_OPEN_LLM_LIST,
+)
+from optimum_benchmark import Benchmark, BenchmarkConfig, BenchmarkReport
+from optimum_benchmark.logging_utils import setup_logging
+
+
+class BenchmarkRunner(ABC):
+    def __init__(self, backend: str, hardware: str, subset: Optional[str] = None, machine: Optional[str] = None):
+        self.backend = backend
+        self.hardware = hardware
+        self.subset = subset or os.getenv("SUBSET", None)
+        self.machine = machine or os.getenv("MACHINE", None)
+        self.logger = getLogger("llm-perf-backend")
+
+        if self.machine is None and self.subset is None:
+            self.push_repo_id = f"optimum-benchmark/llm-perf-{self.backend}-{self.hardware}-debug"
+            self.canonical_pretrained_open_llm_list = ["gpt2"]
+            self.subset = "unquantized"
+        elif self.machine is not None and self.subset is not None:
+            self.push_repo_id = (
+                f"optimum-benchmark/llm-perf-{self.backend}-{self.hardware}-{self.subset}-{self.machine}"
+            )
+        else:
+            raise ValueError("Either both MACHINE and SUBSET should be set for benchmarking or neither for debugging")
+
+        self.attention_configs = self._get_attention_configs()
+        self.weights_configs = self._get_weights_configs(self.subset)
```

Comment:
this is very specific for pytorch backend, I don't see it holding once more backends are introduced